src/OmniSharp.Roslyn.CSharp/Helpers/LocationExtensions.cs

```diff
@@ -13,7 +13,7 @@ public static QuickFix GetQuickFix(this Location location, OmniSharpWorkspace wo
             if (!location.IsInSource)
                 throw new Exception("Location is not in the source tree");
 
-            var lineSpan = location.GetLineSpan();
+            var lineSpan = location.GetMappedLineSpan();
```

Comment:
@filipw So, I tried looking at it. But honestly don't know why Roslyn returns so weird line spans when using `GetMappedLineSpan()` with the Cake file.

I debugged the `ShouldIncludeUsagesFromLoadedFiles()` fact from Cake's `FindUsagesFacts` and flipping this back to `GetLineSpan()` results in this:
![image](https://user-images.githubusercontent.com/7863439/70183656-de8c8500-16ee-11ea-9d2a-f4071309ba3f.png)
While the new change gives this:
![image](https://user-images.githubusercontent.com/7863439/70183730-f6fc9f80-16ee-11ea-8bff-af46e6ef164f.png)

The whole `build.cake` file can be seen [here](https://gist.github.com/mholo65/9e55c33b4d63a7b90c3e37cc183eee32).  The loaded `foo.cake` is basically the same, but from line 8481 onwards, it's
```csharp
#line 1 "C:/src/gh/omnisharp-roslyn/test-assets/test-projects/CakeProject/foo.cake"
Information("Hello World");
```

Could the issue be that we the contents of `foo.cake` is available in both `build.cake` (src included via `#load` directive) and in the document itself. Quickest fix, would be as you already mentioned, to exclude `.cake` files from this mapping, and just do it the old way, but I'd really like to avoid having Cake specific code inside `OmniSharp.Roslyn.CSharp`.

Reply:
Ok, so now I see. The line spans are actually correct (we'll just need to change Cake stuff to not translate it). But the text it fetches here:
https://github.com/OmniSharp/omnisharp-roslyn/blob/8b913fe547f97535c55db64490cbe362a7f4ea6b/src/OmniSharp.Roslyn.CSharp/Helpers/LocationExtensions.cs#L21 doesn't do line span mapping, which results in such weird text being returned. The line span mapping in Cake, I can remove. But how to get the correct text, I don't know.

Reply:
Sent you a PR for the quickest fix here
https://github.com/filipw/omnisharp-roslyn/pull/1